std_execution.bs

```diff
@@ -3022,3 +3039,152 @@ are all well-formed.
+
+          - `execution::set_error(r, e)` is expression-equivalent to `(result_ptr_->emplace<2>(e), continuation_.resume())`.
+
+          - `execution::set_done(r)` is expression-equivalent to `continuation_.promise().unhandled_done().resume()`.
+
+          - `std::tag_invoke(c, r, as...)` is expression-equivalent to `c(p, as...)` when the type of `c` is not one of `execution::set_value_t`, `execution::set_error_t`, or `execution::set_done_t`.
+
+    2. <b><code><i>sender-awaitable</i>::<i>sender-awaitable</i>(S&& s, P& p)</code></b>
+
+        - <i>Effects:</i> initializes `state_` with <code>connect((S&&) s, <i>awaitable-receiver</i>{&result_, coroutine_handle&lt;P>::from_promise(p)})</code>.
+
+    3. <b><code>value_t <i>sender-awaitable</i>::await_resume()</code></b>
+
+        - <i>Effects:</i> equivalent to:
+
+            <pre highlight="c++">
+            if (result_.index()) == 2)
+              rethrow_exception(std::get&lt;2>(result_));
+            if constexpr (!is_void_v&lt;value_t>)
+              return (value_t&&) std::get&lt;1>(result_);
+            </pre>
+
+### `execution::with_awaitable_senders` <b>[execution.coro_utils.with_awaitable_senders]</b> ### {#spec-execution.coro_utils.with_awaitable_senders}
+
+  1. `with_awaitable_senders`, when used as the base class of a coroutine promise type, makes senders awaitable in that coroutine type.
+
+    In addition, it provides a default implementation of `unhandled_done()` such that if a sender completes by calling `execution::set_done`, it is treated as if an uncatchable "done" exception were thrown from the <i>await-expression</i>. In practice, the coroutine is never resumed, and the `unhandled_done` of the coroutine caller's promise type is called.
+
+    <pre highlight="c++">
+    template &lt;class Promise>
+      requires is_class_v&lt;Promise> && same_as&lt;Promise, remove_cvref_t&lt;Promise>>
+      struct with_awaitable_senders {
+        template&lt;OtherPromise>
+          requires (!same_as&lt;OtherPromise, void>)
+        void set_continuation(coroutine_handle&lt;OtherPromise> h) noexcept;
+
+        coroutine_handle&lt;> continuation() const noexcept { return continuation_; }
+
+        coroutine_handle&lt;> unhandled_done() noexcept {
+          return done_handler_(continuation_.address());
+        }
+
+        template&lt;class Value>
+        <i>see-below</i> await_transform(Value&& value);
+
+       private:
+        coroutine_handle&lt;> continuation_{}; // exposition only
+        coroutine_handle&lt;> (*done_handler_)(void*) noexcept = // exposition only
+          [](void*) noexcept -> coroutine_handler&lt;> { terminate(); };
+      };
+    </pre>
+
+  2. <b>`void set_continuation(coroutine_handle<OtherPromise> h) noexcept`</b>
```

Comment:
Do we need to specify that this method is only allowed to be called once?
Otherwise it may be possible to call it once with a coroutine_handle<P1> where P1::unhandled_done() is defined and a subsequent time with coroutine_handle<P2> where P2::unhandled_done() is not defined.
In this case you'd be left with the previous coroutine's done_handler_ as defined here.
Maybe it should always be assigned back to a "default_unhandled_done" function if the promise doesn't support it?